### ISSUE TITLE
`odbc::dbWriteTable` returns error if data.frame to write is over 1024 records log. 

### ISSUE THREAD

**Issue by CerebralMastication** [r-dbi/odbc#217]
### Database
MS SQL Server 2016 (SP2-CU1)

### Reproducible Example

`odbc::dbWriteTable` returns error if data.frame to write is over 1024 records log. 

```{r}
library(odbc)

myuser <- getPass::getPass("enter your username:")
mypwd <- getPass::getPass("enter your password:")

con <- dbConnect(odbc::odbc(), 
                 .connection_string = paste0("Driver={Simba SQL Server ODBC Driver}; 
                                       Server={my_db_here}; 
                                       Database=Analysts; 
                                       UID=renre\\",myuser ,"; 
                                       PWD=",mypwd ,"; 
                                       Integrated Security=NTLM"))

tst <- data.frame(x=rnorm(1025), y=rnorm(1025))
odbc::dbWriteTable(con, 'jal_test_odbc', tst, overwrite=TRUE)
```
which results in the following error (I added CR to make it wrap):
```
> odbc::dbWriteTable(con, 'jal_test_odbc', tst, overwrite=TRUE)
Error in result_insert_dataframe(rs@ptr, values) : 
  nanodbc/nanodbc.cpp:1587: HY000: The incoming tabular data stream (TDS) remote procedure call 
(RPC) protocol stream is incorrect. Parameter 1 (""): The supplied length is not valid for data type 
nvarchar(max). Check the source data for invalid lengths. An example of an invalid length is data of 
nchar type with an odd length in bytes.,  
```
However, if I turn the data.frame down to 1024 records, it runs without error. I've also tried with three columns and the 1024 record constraint still holds. I'm baffled! 

I'm using the Simba ODBC driver for unixODBC on a macbook running High Sierra. 

-->

<details>
<summary>Session Info</summary>

```r
devtools::session_info()

> devtools::session_info()
Session info ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 setting  value                       
 version  R version 3.4.4 (2018-03-15)
 system   x86_64, darwin15.6.0        
 ui       RStudio (1.2.830)           
 language (EN)                        
 collate  en_US.UTF-8                 
 tz       America/New_York            
 date     2018-08-31                  

Packages ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 package    * version    date       source                     
 base       * 3.4.4      2018-03-15 local                      
 bit          1.1-14     2018-05-29 CRAN (R 3.4.4)             
 bit64        0.9-7      2017-05-08 CRAN (R 3.4.0)             
 blob         1.1.1      2018-03-25 CRAN (R 3.4.4)             
 compiler     3.4.4      2018-03-15 local                      
 curl         3.2        2018-03-28 CRAN (R 3.4.4)             
 datasets   * 3.4.4      2018-03-15 local                      
 DBI          1.0.0.9000 2018-08-30 Github (r-dbi/DBI@08e4a31) 
 devtools     1.13.6     2018-06-27 CRAN (R 3.4.4)             
 digest       0.6.16     2018-08-22 CRAN (R 3.4.4)             
 getPass    * 0.2-2      2017-07-21 cran (@0.2-2)              
 git2r        0.23.0     2018-07-17 CRAN (R 3.4.4)             
 graphics   * 3.4.4      2018-03-15 local                      
 grDevices  * 3.4.4      2018-03-15 local                      
 hms          0.4.2      2018-03-10 CRAN (R 3.4.4)             
 httr         1.3.1      2017-08-20 CRAN (R 3.4.1)             
 knitr        1.20       2018-02-20 CRAN (R 3.4.3)             
 memoise      1.1.0      2017-04-21 CRAN (R 3.4.0)             
 methods    * 3.4.4      2018-03-15 local                      
 odbc       * 1.1.6      2018-08-30 Github (r-dbi/odbc@b4c6c3f)
 pkgconfig    2.0.2      2018-08-16 CRAN (R 3.4.4)             
 R6           2.2.2      2017-06-17 CRAN (R 3.4.0)             
 Rcpp         0.12.18    2018-07-23 cran (@0.12.18)            
 rlang        0.2.2      2018-08-16 CRAN (R 3.4.4)             
 rstudioapi   0.7        2017-09-07 CRAN (R 3.4.1)             
 stats      * 3.4.4      2018-03-15 local                      
 tictoc     * 1.0        2014-06-17 CRAN (R 3.4.0)             
 tools        3.4.4      2018-03-15 local                      
 utils      * 3.4.4      2018-03-15 local                      
 withr        2.1.2      2018-03-15 CRAN (R 3.4.4)             

```
</details>

**Comment by crossxwill** [comment]
I'm having the same problem.

**Comment by scwilkins100** [comment]
I've also encountered this issue. 
Only occurs when writing from a linux environment to a Microsoft SQL server using a DSN. 
When writing from a windows environment to the same server there is no issue with row count. 
I've chunked the data and written 1024 rows at a time as a work around.

**Comment by jimhester** [comment]
I can't reproduce this on any of the Simba, freeTDS or Microsoft drivers on macOS, with SQL Server running in a docker container using the devel version of odbc.

**Comment by jimhester** [comment]
Ok strangely enough now I can reproduce it, but only with the Simba drivers. I will try to figure out why and possibly open a Simba ticket.

**Comment by blairj09** [comment]
This seems to have resolved with the latest (1.2.1) release of this package.

**Comment by ejenvey** [comment]
> This seems to have resolved with the latest (1.2.1) release of this package.

Confirmed this update also fixed this issue for me running from a Linux RStudio Server Pro environment to a SQL Server 2016 db.
